pkg/dtest_k3s/k3s.go

```diff
@@ -93,13 +100,18 @@ func isKubeconfigReady(ctx context.Context) bool {
 		return false
 	}
 
-	cmd := dexec.CommandContext(ctx, "docker", "exec", "-i", id, "test", "-e", "/etc/rancher/k3s/k3s.yaml")
-	err := cmd.Start()
+	cmd := dexec.CommandContext(ctx, "docker", "exec", "-i", id, "sh", "-c",
+		"if test -e /etc/rancher/k3s/k3s.yaml; then echo true; else echo false; fi")
+	outBytes, err := cmd.Output()
 	if err != nil {
 		panic(err)
 	}
-	_ = cmd.Wait()
-	return cmd.ProcessState.ExitCode() == 0
+	outString := strings.TrimSpace(string(outBytes))
+	ret, err := strconv.ParseBool(outString)
+	if err != nil {
+		panic(fmt.Errorf("ParseBool: %q: %w", outString, err))
+	}
+	return ret
```

Comment:
This is a key part of "better error handling"; it distinguishes between "`docker exec` failed" and "`docker exec` succeeded, and we're just not ready yet".